File: AngularFront/src/app/common/purchase.ts
import { Address } from "./address";
import { Customer } from "./customer";
import { Order } from "./order";
import { OrderItem } from "./order-item";

export class Purchase {
    customer!: Customer ;
    customerAddress!:Address;
    order!: Order;
    orderItems!:OrderItem[];
    
}

File: src/main/java/pl/szczypkowski/electronicshop/dto/Purchase.java
package pl.szczypkowski.electronicshop.dto;

import lombok.Data;
import pl.szczypkowski.electronicshop.entity.Address;
import pl.szczypkowski.electronicshop.entity.Customer;
import pl.szczypkowski.electronicshop.entity.Order;
import pl.szczypkowski.electronicshop.entity.OrderItem;

import java.util.Set;
@Data
public class Purchase {
    private Customer customer;
    private Address customerAddress;
    private Order order;
    private Set<OrderItem> orderItems;
}
